packages/core/src/awsService/cloudWatchLogs/commands/tailLogGroup.ts

```diff
@@ -83,7 +83,9 @@ async function handleSessionStream(
             }
         }
     } catch (err) {
-        throw new ToolkitError('Caught on-stream exception')
+        throw new ToolkitError(
+            `Encountered LiveTail on-stream failure for session ${session.uri}: ${(err as Error).message}`
```

Comment:
I would suggest not catching the error at all, just let it bubble up. please review our [guidelines](https://github.com/aws/aws-toolkit-vscode/blob/master/docs/CODE_GUIDELINES.md#exceptions).

if you must catch and re-throw this, then it's usually a good idea to use `ToolkitError.chain` so that the original error is part of its [chain](https://github.com/aws/aws-toolkit-vscode/blob/master/docs/arch_develop.md#chaining).